Tests/SonarScanner.MSBuild.Tasks.UnitTests/MoveDirectoryTests.cs

```diff
@@ -0,0 +1,98 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarScanner.MSBuild.Tasks;
+using SonarScanner.MSBuild.Tasks.UnitTests;
+using TestUtilities;
+
+namespace SonarScanner.Integration.Tasks.UnitTests
+{
+    [TestClass]
+    public class MoveDirectoryTests
+    {
+        private MoveDirectory sut;
+
+        public TestContext TestContext { get; set; }
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            sut = new MoveDirectory();
+        }
+
+        [TestMethod]
+        [DataRow(null)]
+        [DataRow("")]
+        [DataRow("   ")]
+        public void MoveDirectory_InvalidSourceDirectoryPath_ReturnsFalse(string sourceDirectory)
+        {
+            // Arrange
+            var dummyEngine = new DummyBuildEngine();
+            sut.BuildEngine = dummyEngine;
+            sut.SourceDirectory = sourceDirectory;
+            sut.DestinationDirectory = @"C:\SomeRandomPath\";
+
+            // Act & Assert
+            sut.Execute().Should().BeFalse();
+            dummyEngine.AssertSingleErrorExists("The source directory is invalid.");
+        }
+
+        [TestMethod]
+        [DataRow(null)]
+        [DataRow("")]
+        [DataRow("   ")]
+        public void MoveDirectory_InvalidDestinationDirectoryPath_ReturnsFalse(string destinationDirectory)
+        {
+            // Arrange
+            var dummyEngine = new DummyBuildEngine();
+            sut.BuildEngine = dummyEngine;
+            sut.SourceDirectory = @"C:\SomeRandomPath\";
+            sut.DestinationDirectory = destinationDirectory;
+
+            // Act & Assert
+            sut.Execute().Should().BeFalse();
+            dummyEngine.AssertSingleErrorExists("The destination directory is invalid.");
+        }
+
+        [TestMethod]
+        public void MoveDirectory_ValidPaths_DirectoryMoved()
+        {
+            // Arrange
+            var root = TestUtils.CreateTestSpecificFolder(TestContext);
+            var sourceDirectory = Path.Combine(root, "Source");
+            var destinationDirectory = Path.Combine(root, "Destination");
+            Directory.CreateDirectory(sourceDirectory);
+            File.WriteAllBytes(Path.Combine(sourceDirectory, "RandomFile.txt"), new byte[] { });
+            var dummyEngine = new DummyBuildEngine();
+            sut.DestinationDirectory = destinationDirectory;
+            sut.SourceDirectory = sourceDirectory;
+            sut.BuildEngine = dummyEngine;
+
+            // Act & Assert
+            sut.Execute().Should().BeTrue();
+            dummyEngine.AssertNoErrors();
+            File.Exists(Path.Combine(destinationDirectory, "RandomFile.txt")).Should().BeTrue();
```

Comment:
Let's move 2 files to be sure all of them are moved. The second one could be in a subdir to prove that this universal task does what we expect him to do.